### PR TITLE
Component reconciler refimpl

### PR DESCRIPTION
**Pull request by tobiscr**

### CODE REVIEW

File: pkg/reconciler/service/reconciler.go [hunk 1]
```diff
@@ -33,23 +33,35 @@ const (
...
+
+var (
+	wsFactory *workspace.Factory //singleton
+	m         sync.Mutex
 )
 
+type ActionContext struct {
+	KubeClient       kubernetes.Client
+	WorkspaceFactory *workspace.Factory
+	Context          context.Context
```
Comment (rJankowski93) [inline pkg/reconciler/service/reconciler.go:47]: I know why you set Context here, but I found this article of creators golang - [context-and-structs](https://blog.golang.org/context-and-structs)

Quote:
Contexts should not be stored inside a struct type, but instead passed to each function that needs it.


It's only suggestion, we can stay with current solution.

**Review by rJankowski93 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -36,3 +36,47 @@ By default, expensive test suites are disabled. To enable them, before you execu
...
+   
+   Example:
+   
+   `pkg/reconciler/instances/new-reconciler.sh istio`
+
+2. The script creates a new package including the boilerplate-code required to initialize a
+   new component reconciler instance during runtime.
+   
+   Please edit the files inside the package:
+
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply README.md:60]: Ordering is not critical actually.

File: README.md [hunk 3]
```diff
@@ -36,3 +36,47 @@ By default, expensive test suites are disabled. To enable them, before you execu
...
+2. The script creates a new package including the boilerplate-code required to initialize a
+   new component reconciler instance during runtime.
+   
+   Please edit the files inside the package:
+   
+      1. Edit the file `action.go` and encapsulate custom reconciliation logic in `Action` structs.
+
+      2. Edit the `$componentName.go` file and
+            1. use the `WithDependencies()` method to list the components which are required before
```
[the remark this answers is not in the document]
Comment (tobiscr) [reply README.md:64]: The "component reconciler" is meant

**Review by rJankowski93 — DISMISSED** [review]

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (NHingerl) [reply README.md:60]: alright then we need the numbering.

**Review by NHingerl — APPROVED** [review]

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**